### ISSUE TITLE
Automatic switching from light to dark mode on macOS has bugs

### ISSUE THREAD

**Issue by LenaWil** [Murmele/Gittyup#408]
macOS has the possibility to automatically switch to dark mode when the sun goes under.

This does not work correctly with Gittyup.

![Schermafbeelding 0004-12-23 om 19 51 41](https://user-images.githubusercontent.com/6208937/209394256-46374b42-722e-4d3a-be16-9b0b81ee5264.png)

**Comment by Murmele** [comment]
@LenaWil currently we don't have a macos developer. Shall this change happen only at restart or when the os chabges to darkmode Gittyup changes too?

**Comment by LenaWil** [comment]
When the sun goes under and the theme of macOS goes to dark, and Gittyup is open at the time of the change, Gittyup changes too and the glitch happens. 

I tried to reproduce it both in Gnome on Ubuntu and on Windows 11 by changing the theme while it was open there, but the same glitch doesn’t happen there, because Gittyup stayed the same theme while opened even when set to system.

**Comment by Murmele** [comment]
Thank you. We will check out to get it work also with Gnome so this might fix then also your problem

**Comment by LenaWil** [comment]
> Thank you. We will check out to get it work also with Gnome so this might fix then also your problem
Okay, just to be clear, it did _not_ happen with gnome.

**Comment by Murmele** [comment]
Yes but if we get it work there we might also fix the issue with MacOS. Do you have screenshots how it looks in the lightmode? What happens if you start Gittyup at night, does it look same?

**Comment by LenaWil** [comment]
If I start up Gitty in light mode: 
![Light mode](https://user-images.githubusercontent.com/6208937/211212421-323a2a33-b78a-45e9-a740-4cb49959707f.png)

And this is what dark mode looks like normally:
![Dark mode](https://user-images.githubusercontent.com/6208937/211212463-0942b178-ef8a-4328-a9cd-762de1e8415d.png)

The inverse glitch: (So switching from dark to light, aka sun rise, while open)
![Dark to light mode](https://user-images.githubusercontent.com/6208937/211212540-eeb42cfc-4279-48fc-b76a-70c84be056dd.png)

**Comment by Murmele** [comment]
Thanks for the screenshot. I will have a look into it

**Comment by Murmele** [comment]
Just for documentation:
Used in Application.cpp
````cpp
  // Initialize theme.
  mTheme.reset(Theme::create(parser.value("theme")));
  setStyle(mTheme->style());
  setStyleSheet(mTheme->styleSheet());
  ````

**Comment by Murmele** [comment]
@LenaWil If you restart Gittyup when this problem occurs, are the colors correct?

**Comment by LenaWil** [comment]
Yes.

**Comment by rokejulianlockhart** [comment]
Same on KDE Plasma.

![image](https://user-images.githubusercontent.com/42837531/235324328-73af5c9b-fce1-40cb-81ca-b337b53f5271.png)
